solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/action/DeleteStaleChildren.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.data.ingest.solr.action;
+
+import static edu.unc.lib.dl.fcrepo4.RepositoryPaths.getContentRootPid;
+
+import java.util.Map;
+
+import org.apache.solr.common.StringUtils;
+import org.springframework.util.Assert;
+
+import edu.unc.lib.dl.data.ingest.solr.SolrUpdateRequest;
+import edu.unc.lib.dl.data.ingest.solr.exception.IndexingException;
+import edu.unc.lib.dl.search.solr.model.BriefObjectMetadata;
+import edu.unc.lib.dl.search.solr.util.SearchFieldKeys;
+import edu.unc.lib.dl.search.solr.util.SolrSettings;
+
+/**
+ * Deletes solr records for all objects which are children of the specified
+ * object if they have not been updated since the time specified by
+ * staleTimestamp
+ *
+ * @author bbpennel
+ *
+ */
+public class DeleteStaleChildren extends AbstractIndexingAction {
+
+    public static final String STALE_TIMESTAMP = "staleTimestamp";
+
+    public DeleteStaleChildren() {
+    }
+
+    @Override
+    public void performAction(SolrUpdateRequest updateRequest) throws IndexingException {
+        Assert.notEmpty(updateRequest.getParams());
+
+        Map<String, String> params = updateRequest.getParams();
+        String staleTimestamp = params.get(STALE_TIMESTAMP);
+        if (StringUtils.isEmpty(staleTimestamp)) {
```

Comment:
why not use the spring framework again for consistency, e.g., Assert.notNull(staleTimestamp)?

Reply:
This was mainly copy and pasted from code before i started using assertions, it should be updated now.